include/envoy/http/header_map.h

```diff
@@ -65,9 +71,10 @@ class LowerCaseString {
   void lower() {
     std::transform(string_.begin(), string_.end(), string_.begin(), absl::ascii_tolower);
   }
-  bool valid() const { return validHeaderString(string_); }
+  bool valid() const { return validHeaderString(string_) && validHeaderString(original_string_); }
```

Comment:
They're not gonna want to double the validation for people that haven't opted in